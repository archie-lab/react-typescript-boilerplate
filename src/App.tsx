import React, {FC, Suspense} from "react";
import {hot} from "react-hot-loader/root";
import "./App.scss";
import {BrowserRouter, Route, Switch} from "react-router-dom";
import {useTranslation} from "react-i18next";
import {useTypedSelector} from "./store";

const App: FC<Props> = () => {
  const message = useTypedSelector((state) => state.common.message);
  const {t} = useTranslation();
  return (
    <BrowserRouter>
      <Suspense fallback={<p>Error</p>}>
        <Switch>
          <Route
            exact
            pat="/"
            component={(): JSX.Element => <h1 className="app">{`${t("title")}: ${message}` || "Hello, World!"}</h1>}
          />
        </Switch>
      </Suspense>
    </BrowserRouter>
  );
};

interface Props {
  message?: string;
}

export default hot<FC<Props>>(App);
