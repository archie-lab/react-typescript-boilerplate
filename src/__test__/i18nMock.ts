import i18n from "i18next";
import LanguageDetector from "i18next-browser-languagedetector";
import {initReactI18next} from "react-i18next";

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    fallbackLng: "en",

    // have a common namespace used around the full app
    ns: ["common"],
    defaultNS: "common",

    debug: false,

    interpolation: {
      escapeValue: false // not needed for react!!
    },
    resources: {
      en: {}
    }
  });
