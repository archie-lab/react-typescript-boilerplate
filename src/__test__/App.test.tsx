import * as React from "react";
import {shallow} from "enzyme";
// import Adapter from "enzyme-adapter-react-16";
import App from "../App";
import "./i18nMock";

// configure({adapter: new Adapter()});

jest.mock("react-redux", () => ({
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  useDispatch: () => {},
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  useSelector: jest.fn().mockImplementation((selector) => ({qwe: "qwe"}))
}));

describe("App Component", () => {
  const spy = jest.spyOn(global.console, "error");
  it("should exist", () => {
    expect(spy).not.toHaveBeenCalled();
    expect(shallow(<App />)).toBeTruthy();
  });

  it("should match snapshot", () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
