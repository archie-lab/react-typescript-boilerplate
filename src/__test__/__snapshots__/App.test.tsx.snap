// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `
<BrowserRouter>
  <Suspense
    fallback={
      <p>
        Error
      </p>
    }
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        pat="/"
      />
    </Switch>
  </Suspense>
</BrowserRouter>
`;
